src/server/lib/game/MessageExchanger.hh

```diff
@@ -31,9 +32,9 @@ class MessageExchanger
   void pushMessage(IMessagePtr message);
 
   private:
-  NetworkMessageQueuePtr m_inputMessageQueue{};
-  IMessageQueuePtr m_internalMessageQueue{};
   IMessageQueuePtr m_outputMessageQueue{};
+  IMessageQueuePtr m_internalMessageQueue{};
+  NetworkMessageQueuePtr m_inputMessageQueue{};
```

Comment:
This was the second problem. According to [this link](https://stackoverflow.com/questions/2254263/order-of-member-constructor-and-destructor-calls) the member variables are destroyed in reverse order of their construction. It means that we destroyed first the output message queue and then the internal message queue.
But it could very well be that while we destroyed the output message queue one message in the internal queue was processed and was producing a message to send to the output message queue which was not valid anymore.